cereus/src/main/java/com/cactus/adapters/AuthLogin.java

```diff
@@ -0,0 +1,186 @@
+ * that stores our Users and Grocery Lists.
+ */
+public class AuthLogin implements  AuthAdapter{
+
+    /**
+     * Returns a User object that corresponds to the provided username and password.
+     *
+     * The username and password is sent to a server to return the User corresponding to the username and password.
+     *
+     * If the username and password do not correspond to an existing User,
+     * then null is returned, and no login takes place.
+     *
+     * Note: the password is not present in the User, and a null value is in place
+     *
+     * @param username a String containing the username of the user to be logged in
+     * @param password a String containing the password to validate
+     * @return         a User object that corresponds to the given parameters
+     *
+     * @see User
+     */
+    @Override
+    public User login(String username, String password) {
+        HashMap<String, String> login = new HashMap<>();
+        login.put("username", username);
+        login.put("password", password);
+        URI uri;
+        try{
+            uri = new URIBuilder("http://localhost:8080/login").addParameter("username",
+                username).addParameter("password", password).build();
+            }
+        catch(URISyntaxException u) {
+            return null;
+        }
+        User user;
+        try {
+            user = sendRequest(login, uri);
+        }
+        catch(IOException | InterruptedException i){
+            return null;
+        }
+        if (user.getToken() == null) {
+            return null;
+        }
+        return user;
+
+    }
+
+    /**
+     * Returns a User object created with the provided username, password, and name.
+     *
+     * Note: the resulting user object is also logged in as the current user, and no further call to log in the User
+     * is required
+     *
+     * The username, name and password is sent to a server to check if the username doesn't correspond to an existing
+     * user.
+     *
+     * If the user has the same username and password (but different name)
+     * as one that already exists, then that person is logged in and no creation takes place.
+     *
+     * If the user has the same username and different password as an existing user,
+     * then null is returned and no creation or login takes place.
+     *
+     * If the username has a different username, then a User object corresponding to the given information is created,
+     * and returned.
+     *
+     * Note: the password is not present in the User, and a null value is in place
+     *
+     * @param username a String containing the username of the new user
+     * @param password a String containing the password of the new user
+     * @param name     a String containing the name of the user
+     * @return         a User object created with the given parameters
+     *
+     * @see User
+     */
+    @Override
+    public User create(String username, String password, String name) {
```

Comment:
While the documentation for this method matches its implementation, note that the register user page redirects to the login page on successful user creation. This means that the functionality where a user is also logged in on registration is undesirable. We want registration and login to be separate processes, so this method should _only_ deal with registration.

Reply:
I'll separate the two methods